### PR TITLE
Move git checks to WestCommand

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
This makes it easier for commands to check the git version.

### CODE REVIEW

File: src/west/commands.py [hunk 1]
```diff
@@ -224,6 +226,29 @@ def check_output(self, args, cwd=None):
         log.dbg(f"running '{cmd_str}' in {cwd or os.getcwd()}",
                 level=log.VERBOSE_VERY)
         return subprocess.check_output(args, cwd=cwd)
+
+    def git_version_at_least(self, version):
```
Comment (tejlmand) [inline src/west/commands.py:238]: minor, should we call it `git_version_minimum_required` / `git_version_minimum`?

**Review by tejlmand — APPROVED** [review]
Looks good. Nice cleanup.
Just a minor suggestion.

**Review by marc-hb — COMMENTED** [review]
Why are `check_call` and `check_output` methods at all?

```
pylint
R0201: Method could be a function (no-self-use)
```

File: src/west/commands.py (hunk 1 above)
Comment (mbolivar-nordic) [reply src/west/commands.py]: > minor, should we call it `git_version_minimum_required` / `git_version_minimum`?

I changed it to a property, `git_version_info`, following along with `sys.version_info`.

**Comment by mbolivar-nordic** [comment]
> Why are `check_call` and `check_output` methods at all?

Because I hate `west.log` and want to convert all the output routines (`log.die`, etc.) to WestCommand methods (`WestCommand.die`, etc.) at some point.

That will remove the need for global state in `west.log` (continuing work towards resolving #149) without sacrificing the ability to do things like `west --quiet update` by adding a per-instance "be quiet" flag or so.

Moved to staticmethod for now to silence the linter.

**Review by marc-hb — APPROVED** [review]
> Because I hate west.log and want to convert all the output routines ...

I don't get all the details but it's more than enough for me to know you thought about it.

> Moved to staticmethod for now to silence the linter.

It's not just the linter, I find `@staticmethod` more visible. I often skip the list of arguments.

The new `git_version_info()` looks better and great; much more flexible.

**Pull request merged**